ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcParserHandler.java

```diff
@@ -39,6 +43,10 @@ public static Handler<RoutingContext> handler() {
         try {
           ctx.put(ContextKey.REQUEST_BODY_AS_JSON_OBJECT.name(), ctx.getBodyAsJson());
```

Comment:
It is not related to this PR, but I think we should harmonize both implementations. Instead of always trying de deserialize un Object and fallback to array, check if is a JsonObject or JsonArray before trying to deserialize it.
From my check, `isObject()` and `isArray()` are already calculated so there shouldn't any a lot of overhead on Object use case and should improve Array use case.

Reply:
On 2). we need to measure it to be sure. I think even payloads close to the limit can suffer from the extra serialization.

Reply:
As discussed, I went back to using a global setting (if engine API enabled).
This setting will apply to both JSON-RPC and Engine-JSON-RPC